internal/service/outposts/outpost_assets_data_source.go

```diff
@@ -36,6 +64,15 @@ func DataSourceOutpostAssetsRead(d *schema.ResourceData, meta interface{}) error
 	input := &outposts.ListAssetsInput{
 		OutpostIdentifier: outpost_id,
 	}
+
+	if _, ok := d.GetOk("host_id_filter"); ok {
+		input.HostIdFilter = flex.ExpandStringList(d.Get("host_id_filter").([]interface{}))
+	}
+
+	if _, ok := d.GetOk("status_id_filter"); ok {
+		input.StatusFilter = flex.ExpandStringList(d.Get("status_id_filter").([]interface{}))
```

Comment:
If you change to TypeSet in the schema, update to `flex.ExpandStringSet` here

```suggestion
		input.StatusFilter = flex.ExpandStringSet(d.Get("status_id_filter").(*schema.Set))
```